rest/tests/test_course.py

```diff
@@ -85,3 +256,95 @@ def test_get_courses(self):
 
         # anonymous
         self.get_courses(repo_slug=self.repo.slug, expected_status=403)
+
+    def test_get_course(self):
+        """
+        Authorizations for get a course
+        Any user of the repo can see courses
+        """
+        # administrator of the repo (already logged in)
+        self.get_course(repo_slug=self.repo.slug, course_id=self.course.id)
+        self.logout()
+
+        # curator
+        self.login(self.curator_user.username)
+        self.get_course(repo_slug=self.repo.slug, course_id=self.course.id)
+        self.logout()
+
+        # author
+        self.login(self.author_user.username)
+        self.get_course(repo_slug=self.repo.slug, course_id=self.course.id)
+        self.logout()
+
+        # user without repo
+        self.login(self.user_norepo)
+        self.get_course(
+            repo_slug=self.repo.slug,
+            course_id=self.course.id,
+            expected_status=403
+        )
+        self.logout()
+
+        # anonymous
+        self.get_course(
+            repo_slug=self.repo.slug,
+            course_id=self.course.id,
+            expected_status=403
+        )
+
+    def test_delete_course(self):
+        """
+        Authorizations for delete a course
+        Only user with import permission (admin and curator) can delete a repo
+        """
+        course_params = {
+            'org': 'my org',
+            'repo_id': self.repo.id,
+            'course_number': 'second course number',
+            'run': "second course run",
+            'user_id': self.user.id
+        }
+
+        # anonymous user
+        self.logout()
+        self.delete_course(
+            self.repo.slug,
+            self.course.id,
+            expected_status=403
+        )
+
+        # user not belonging to repo
+        self.login(self.user_norepo.username)
+        self.delete_course(
+            self.repo.slug,
+            self.course.id,
+            expected_status=403
+        )
+        self.logout()
+
+        # author
+        self.login(self.author_user.username)
+        self.delete_course(
+            self.repo.slug,
+            self.course.id,
+            expected_status=403
+        )
+        self.logout()
+
+        # curator
+        self.login(self.curator_user.username)
+        self.delete_course(
+            self.repo.slug,
+            self.course.id
+        )
+        self.logout()
+
+        # recreate course
+        self.course = create_course(**course_params)
+
+        # administrator
+        self.login(self.user.username)
+        self.delete_course(
+            self.repo.slug,
+            self.course.id
+        )
```

Comment:
Can you add a test for PATCH and PUT on a course? It would be fine to just assert a 405 status code since they're not implemented, no need to get complicated.